x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/ExpressionTests.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+package org.elasticsearch.xpack.eql.parser;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Literal;
+import org.elasticsearch.xpack.ql.expression.UnresolvedAttribute;
+import org.elasticsearch.xpack.ql.expression.function.UnresolvedFunction;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.And;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.Or;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Neg;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.Equals;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.NotEquals;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ExpressionTests extends ESTestCase {
+
+    private final EqlParser parser = new EqlParser();
+
+    public Expression expr(String source) {
+        return parser.createExpression(source);
+    }
+
+    public void testStrings() throws Exception {
+        assertEquals("hello\"world", AstBuilder.unquoteString("'hello\"world'"));
```

Comment:
1. you could import static `unquoteString` and further mroe extract it a public static method into StringUtils and then invoke it inside AstBuilder.

2. the import should be on IdentifierBuilder.unquoteString` not `AstBuidler` - not sure what IDE you are using but normally you should get a warning about this (static methods are not inherited and thus there's no dispatch).

3. unquote tests should sit in their own test class `IdentifierTests` or `StringUtilsTests`.

Reply:
1 - I can move that to intelliJ
2 - that'll change the import. LiteralBuilder I think would've made the most since. and aren't static methods are still inherited but can't be overwritten, which is why there's no dispatch?
3 - will do

Reply:
static methods by definition are not virtual so the inheritance is really about visibility.
Using a subclass is misleading as the caller might think this is an overridden method when in fact that's not the case. The end result is the same though (it might very well be that IntelliJ has a working for this somewhere that needs enabling).